core/trino-main/src/main/java/io/trino/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -371,6 +373,42 @@ public TableWriterNode map(TableWriterNode node, PlanNode source, PlanNodeId new
                 node.getStatisticsAggregationDescriptor().map(descriptor -> descriptor.map(this::map)));
     }
 
+    public MergeNode map(MergeNode node, PlanNode source)
+    {
+        return map(node, source, node.getId());
+    }
+
+    public MergeNode map(MergeNode node, PlanNode source, PlanNodeId newId)
+    {
+        // Intentionally does not use mapAndDistinct on columns as that would remove columns
+        List<Symbol> newOutputs = map(node.getOutputSymbols());
+
+        return new MergeNode(
+                node.getId(),
+                source,
+                node.getTarget(),
+                node.getTableScanId(),
+                map(node.getProjectedSymbols()),
+                node.getPartitioningScheme().map(partitioningScheme -> map(partitioningScheme, source.getOutputSymbols())),
+                newOutputs);
+    }
+
+    public DeleteAndInsertNode map(DeleteAndInsertNode node, PlanNode source)
+    {
+        return map(node, source, node.getId());
+    }
+
+    public DeleteAndInsertNode map(DeleteAndInsertNode node, PlanNode source, PlanNodeId newId)
```

Comment:
This method is not used otherwise than by the method above. Please inline it.

Reply:
Inlined.